include/picongpu/plugins/binning/BinningFunctors.hpp

```diff
@@ -59,21 +58,53 @@ namespace picongpu
                         // This assumes n_bins and getBinIdx exist
                         validIdx
                             = (
-                                   [&]
+                                   [&](auto const& tupleArg)
                                    {
-                                       auto [isValid, binIdx] = pmacc::memory::tuple::get<1>(tupleArgs).getBinIdx(domainInfo, worker, particle);
-                                       binsDataspace[pmacc::memory::tuple::get<0>(tupleArgs)] = binIdx;
-                                       return isValid;
-                                   }()
+                                    if constexpr(pmacc::memory::tuple::tuple_size_v<decltype(userFunctorData)> > 0)
+                                        {
+                                           auto [isValid, binIdx] = binning::apply(
+                                               [&](auto const&... userFunctorData)
+                                               { return pmacc::memory::tuple::get<1>(tupleArg).getBinIdx(worker, domainInfo, particle, userFunctorData...); },
+                                               userFunctorData);
+
+                                           binsDataspace[pmacc::memory::tuple::get<0>(tupleArg)] = binIdx;
+                                           return isValid;
+                                       }
+                                       else{
+                                           auto [isValid, binIdx] = pmacc::memory::tuple::get<1>(tupleArg).getBinIdx(worker, domainInfo, particle);
+                                           binsDataspace[pmacc::memory::tuple::get<0>(tupleArg)] = binIdx;
+                                           return isValid;
+                                       }
+                                   }(tupleArgs)
                                && ...);
                     },
                     axes);
 
                 if(validIdx)
                 {
                     auto const idxOneD = pmacc::math::linearize(extents, binsDataspace);
-                    DepositionType depositVal = quantityFunctor(worker, particle);
-                    alpaka::atomicAdd(worker.getAcc(), &(histBox[idxOneD]), depositVal, ::alpaka::hierarchy::Blocks{});
+                    if constexpr(pmacc::memory::tuple::tuple_size_v < decltype(userFunctorData) >> 0)
```

Comment:
That's what's meant, right?
```suggestion
                    if constexpr(pmacc::memory::tuple::tuple_size_v<decltype(userFunctorData)> > 0)
```

Reply:
Yes, but the auto formatting seems broken. Maybe we need to update the clang-format version we use